core/src/main/java/io/questdb/griffin/engine/functions/catalogue/MatViewsFunctionFactory.java

```diff
@@ -246,7 +253,7 @@ public CharSequence getStrA(int col) {
                         case COLUMN_TABLE_DIR_NAME:
                             return viewDefinition.getMatViewToken().getDirName();
                         case COLUMN_VIEW_STATUS:
-                            return invalid ? "invalid" : "valid";
+                            return lastRefreshStartTimestamp > lastRefreshFinishTimestamp ? "refreshing" : invalid ? "invalid" : "valid";
```

Comment:
```suggestion
                            return invalid ? "invalid" : lastRefreshStartTimestamp > lastRefreshFinishTimestamp ? "refreshing"  : "valid";
```

Reply:
in case of invalidation, lastRefreshStartTimestamp == lastRefreshFinishTimestamp

Reply:
That's a fragile invariant that can change in future. Checking the flag is less error-prone.